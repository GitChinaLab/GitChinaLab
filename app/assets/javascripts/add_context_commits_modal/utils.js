export const findCommitIndex = (commits, commitShortId) => {
  return commits.findIndex((commit) => commit.short_id === commitShortId);
};

export const setCommitStatus = (commits, commitIndex, selected) => {
  const tempCommits = [...commits];
  tempCommits[commitIndex] = {
    ...tempCommits[commitIndex],
    isSelected: selected,
  };
  return tempCommits;
};

export const removeIfReadyToBeRemoved = (toRemoveCommits, commitShortId) => {
  const tempToRemoveCommits = [...toRemoveCommits];
  const isPresentInToRemove = tempToRemoveCommits.indexOf(commitShortId);
  if (isPresentInToRemove !== -1) {
    tempToRemoveCommits.splice(isPresentInToRemove, 1);
  }

  return tempToRemoveCommits;
};

export const removeIfPresent = (selectedCommits, commitShortId) => {
  const tempSelectedCommits = [...selectedCommits];
  const selectedCommitsIndex = findCommitIndex(tempSelectedCommits, commitShortId);
  if (selectedCommitsIndex !== -1) {
    tempSelectedCommits.splice(selectedCommitsIndex, 1);
  }

  return tempSelectedCommits;
};
