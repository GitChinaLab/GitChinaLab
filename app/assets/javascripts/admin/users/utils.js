export const generateUserPaths = (paths, id) => {
  return Object.fromEntries(
    Object.entries(paths).map(([action, genericPath]) => {
      return [action, genericPath.replace('id', id)];
    }),
  );
};
