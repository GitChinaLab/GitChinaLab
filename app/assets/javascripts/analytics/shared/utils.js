import dateFormat from 'dateformat';
import { urlQueryToFilter } from '~/vue_shared/components/filtered_search_bar/filtered_search_utils';
import { dateFormats } from './constants';

export const filterBySearchTerm = (data = [], searchTerm = '', filterByKey = 'name') => {
  if (!searchTerm?.length) return data;
  return data.filter((item) => item[filterByKey].toLowerCase().includes(searchTerm.toLowerCase()));
};

export const toYmd = (date) => dateFormat(date, dateFormats.isoDate);

/**
 * Takes a url and extracts query parameters used for the shared
 * filter bar
 *
 * @param {string} url The URL to extract query parameters from
 * @returns {Object}
 */
export const extractFilterQueryParameters = (url = '') => {
  const {
    source_branch_name = null,
    target_branch_name = null,
    author_username = null,
    milestone_title = null,
    assignee_username = [],
    label_name = [],
  } = urlQueryToFilter(url);

  return {
    selectedSourceBranch: source_branch_name,
    selectedTargetBranch: target_branch_name,
    selectedAuthor: author_username,
    selectedMilestone: milestone_title,
    selectedAssigneeList: assignee_username,
    selectedLabelList: label_name,
  };
};

/**
 * Takes a url and extracts sorting and pagination query parameters into an object
 *
 * @param {string} url The URL to extract query parameters from
 * @returns {Object}
 */
export const extractPaginationQueryParameters = (url = '') => {
  const { sort, direction, page } = urlQueryToFilter(url);
  return {
    sort: sort?.value || null,
    direction: direction?.value || null,
    page: page?.value || null,
  };
};

export const getDataZoomOption = ({
  totalItems = 0,
  maxItemsPerPage = 40,
  dataZoom = [{ type: 'slider', bottom: 10, start: 0 }],
}) => {
  if (totalItems <= maxItemsPerPage) {
    return {};
  }

  const intervalEnd = Math.ceil((maxItemsPerPage / totalItems) * 100);

  return dataZoom.map((item) => {
    return {
      ...item,
      end: intervalEnd,
    };
  });
};
