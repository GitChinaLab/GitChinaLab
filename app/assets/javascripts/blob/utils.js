import Editor from '~/editor/source_editor';

export function initSourceEditor({ el, ...args }) {
  const editor = new Editor({
    scrollbar: {
      alwaysConsumeMouseWheel: false,
    },
  });

  return editor.createInstance({
    el,
    ...args,
  });
}

export default () => ({});
