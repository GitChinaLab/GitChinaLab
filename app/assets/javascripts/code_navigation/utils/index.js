export const cachedData = new Map();

export const getCurrentHoverElement = () => cachedData.get('current');
export const setCurrentHoverElement = (el) => cachedData.set('current', el);

export const addInteractionClass = (path, d) => {
  const lineNumber = d.start_line + 1;
  const lines = document
    .querySelector(`[data-path="${path}"]`)
    .querySelectorAll(`.blob-content #LC${lineNumber}, .line_content:not(.old) #LC${lineNumber}`);
  if (!lines?.length) return;

  lines.forEach((line) => {
    let charCount = 0;
    const el = [...line.childNodes].find(({ textContent }) => {
      if (charCount === d.start_char) return true;
      charCount += textContent.length;
      return false;
    });

    if (el) {
      el.setAttribute('data-char-index', d.start_char);
      el.setAttribute('data-line-index', d.start_line);
      el.classList.add('cursor-pointer', 'code-navigation', 'js-code-navigation');
      el.closest('.line').classList.add('code-navigation-line');
    }
  });
};
