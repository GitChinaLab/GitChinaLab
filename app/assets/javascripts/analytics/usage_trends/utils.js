import { masks } from 'dateformat';
import { get } from 'lodash';
import { formatDate } from '~/lib/utils/datetime_utility';

const { isoDate } = masks;

/**
 * Takes an array of items and returns one item per month with the average of the `count`s from that month
 * @param  {Array} items
 * @param  {Number} items[index].count value to be averaged
 * @param  {String} items[index].recordedAt item dateTime time stamp to be collected into a month
 * @param  {Object} options
 * @param  {Object} options.shouldRound an option to specify whether the retuned averages should be rounded
 * @return {Array} items collected into [month, average],
 * where month is a dateTime string representing the first of the given month
 * and average is the average of the count
 */
export function getAverageByMonth(items = [], options = {}) {
  const { shouldRound = false } = options;
  const itemsMap = items.reduce((memo, item) => {
    const { count, recordedAt } = item;
    const date = new Date(recordedAt);
    const month = formatDate(new Date(date.getFullYear(), date.getMonth(), 1), isoDate);
    if (memo[month]) {
      const { sum, recordCount } = memo[month];
      return { ...memo, [month]: { sum: sum + count, recordCount: recordCount + 1 } };
    }

    return { ...memo, [month]: { sum: count, recordCount: 1 } };
  }, {});

  return Object.keys(itemsMap).map((month) => {
    const { sum, recordCount } = itemsMap[month];
    const avg = sum / recordCount;
    if (shouldRound) {
      return [month, Math.round(avg)];
    }

    return [month, avg];
  });
}

/**
 * Takes an array of usage counts and returns the last item in the list
 * @param  {Array} arr array of usage counts in the form { count: Number, recordedAt: date String }
 * @return {String} the 'recordedAt' value of the earliest item
 */
export const getEarliestDate = (arr = []) => {
  const len = arr.length;
  return get(arr, `[${len - 1}].recordedAt`, null);
};

/**
 * Takes an array of queries and produces an object with the query identifier as key
 * and a supplied defaultValue as its value
 * @param  {Array} queries array of chart query configs,
 *                 see ./analytics/usage_trends/components/charts_config.js
 * @param  {any}   defaultValue value to set each identifier to
 * @return {Object} key value pair of the form { queryIdentifier: defaultValue }
 */
export const generateDataKeys = (queries, defaultValue) =>
  queries.reduce(
    (acc, { identifier }) => ({
      ...acc,
      [identifier]: defaultValue,
    }),
    {},
  );
