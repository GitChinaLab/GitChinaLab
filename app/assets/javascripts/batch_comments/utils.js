import { getFormData } from '~/diffs/store/utils';

export const getDraftReplyFormData = (data) => ({
  endpoint: data.notesData.draftsPath,
  data,
});

export const getDraftFormData = (params) => ({
  endpoint: params.notesData.draftsPath,
  data: getFormData(params),
});

export const parallelLineKey = (line, side) => (line[side] ? line[side].line_code : '');

export const showDraftOnSide = (line, side) => {
  // inline mode
  if (side === null) {
    return true;
  }

  // parallel
  if (side === 'left' || side === 'right') {
    const otherSide = side === 'left' ? 'right' : 'left';
    const thisCode = (line[side] && line[side].line_code) || '';
    const otherCode = (line[otherSide] && line[otherSide].line_code) || '';

    // either the lineCodes are different
    // or if they're the same, only show on the left side
    if (thisCode !== otherCode || (side === 'left' && thisCode === otherCode)) {
      return true;
    }
  }

  return false;
};
