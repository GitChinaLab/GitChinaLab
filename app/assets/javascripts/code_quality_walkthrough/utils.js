import { TRACKING_CONTEXT_SCHEMA } from '~/experimentation/constants';
import { getExperimentData } from '~/experimentation/utils';
import { setCookie, getCookie } from '~/lib/utils/common_utils';
import { getParameterByName } from '~/lib/utils/url_utility';
import Tracking from '~/tracking';
import { EXPERIMENT_NAME } from './constants';

export function getExperimentSettings() {
  return JSON.parse(getCookie(EXPERIMENT_NAME) || '{}');
}

export function setExperimentSettings(settings) {
  setCookie(EXPERIMENT_NAME, settings);
}

export function isWalkthroughEnabled() {
  return getParameterByName(EXPERIMENT_NAME);
}

export function track(action) {
  const { data } = getExperimentSettings();

  if (data) {
    Tracking.event(EXPERIMENT_NAME, action, {
      context: {
        schema: TRACKING_CONTEXT_SCHEMA,
        data,
      },
    });
  }
}

export function startCodeQualityWalkthrough() {
  const data = getExperimentData(EXPERIMENT_NAME);

  if (data) {
    setExperimentSettings({ data });
  }
}
