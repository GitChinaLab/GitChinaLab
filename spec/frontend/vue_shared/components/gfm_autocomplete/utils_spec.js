import { escape, last } from 'lodash';
import { GfmAutocompleteType, tributeConfig } from '~/vue_shared/components/gfm_autocomplete/utils';

describe('gfm_autocomplete/utils', () => {
  describe('emojis config', () => {
    const emojisConfig = tributeConfig[GfmAutocompleteType.Emojis].config;
    const emoji = 'raised_hands';

    it('uses : as the trigger', () => {
      expect(emojisConfig.trigger).toBe(':');
    });

    it('searches using the emoji name', () => {
      expect(emojisConfig.lookup(emoji)).toBe(emoji);
    });

    it('limits the number of rendered items to 100', () => {
      expect(emojisConfig.menuItemLimit).toBe(100);
    });

    it('shows the emoji name and icon in the menu item', () => {
      expect(emojisConfig.menuItemTemplate({ original: emoji })).toMatchSnapshot();
    });

    it('inserts the emoji name on autocomplete selection', () => {
      expect(emojisConfig.selectTemplate({ original: emoji })).toBe(`:${emoji}:`);
    });
  });

  describe('issues config', () => {
    const issuesConfig = tributeConfig[GfmAutocompleteType.Issues].config;
    const groupContextIssue = {
      iid: 987654,
      reference: 'gitlab#987654',
      title: "Group context issue title <script>alert('hi')</script>",
    };
    const projectContextIssue = {
      id: null,
      iid: 123456,
      time_estimate: 0,
      title: "Project context issue title <script>alert('hi')</script>",
    };

    it('uses # as the trigger', () => {
      expect(issuesConfig.trigger).toBe('#');
    });

    it('searches using both the iid and title', () => {
      expect(issuesConfig.lookup(projectContextIssue)).toBe(
        `${projectContextIssue.iid}${projectContextIssue.title}`,
      );
    });

    it('limits the number of rendered items to 100', () => {
      expect(issuesConfig.menuItemLimit).toBe(100);
    });

    it('shows the reference and title in the menu item within a group context', () => {
      expect(issuesConfig.menuItemTemplate({ original: groupContextIssue })).toMatchSnapshot();
    });

    it('shows the iid and title in the menu item within a project context', () => {
      expect(issuesConfig.menuItemTemplate({ original: projectContextIssue })).toMatchSnapshot();
    });

    it('inserts the reference on autocomplete selection within a group context', () => {
      expect(issuesConfig.selectTemplate({ original: groupContextIssue })).toBe(
        groupContextIssue.reference,
      );
    });

    it('inserts the iid on autocomplete selection within a project context', () => {
      expect(issuesConfig.selectTemplate({ original: projectContextIssue })).toBe(
        `#${projectContextIssue.iid}`,
      );
    });
  });

  describe('labels config', () => {
    const labelsConfig = tributeConfig[GfmAutocompleteType.Labels].config;
    const labelsFilter = tributeConfig[GfmAutocompleteType.Labels].filterValues;
    const label = {
      color: '#123456',
      textColor: '#FFFFFF',
      title: `bug <script>alert('hi')</script>`,
      type: 'GroupLabel',
    };
    const singleWordLabel = {
      color: '#456789',
      textColor: '#DDD',
      title: `bug`,
      type: 'GroupLabel',
    };
    const numericalLabel = {
      color: '#abcdef',
      textColor: '#AAA',
      title: 123456,
      type: 'ProjectLabel',
    };

    it('uses ~ as the trigger', () => {
      expect(labelsConfig.trigger).toBe('~');
    });

    it('searches using `title`', () => {
      expect(labelsConfig.lookup).toBe('title');
    });

    it('limits the number of rendered items to 100', () => {
      expect(labelsConfig.menuItemLimit).toBe(100);
    });

    it('shows the title in the menu item', () => {
      expect(labelsConfig.menuItemTemplate({ original: label })).toMatchSnapshot();
    });

    it('inserts the title on autocomplete selection', () => {
      expect(labelsConfig.selectTemplate({ original: singleWordLabel })).toBe(
        `~${escape(singleWordLabel.title)}`,
      );
    });

    it('inserts the title enclosed with quotes on autocomplete selection when the title is numerical', () => {
      expect(labelsConfig.selectTemplate({ original: numericalLabel })).toBe(
        `~"${escape(numericalLabel.title)}"`,
      );
    });

    it('inserts the title enclosed with quotes on autocomplete selection when the title contains multiple words', () => {
      expect(labelsConfig.selectTemplate({ original: label })).toBe(`~"${escape(label.title)}"`);
    });

    describe('filter', () => {
      const collection = [label, singleWordLabel, { ...numericalLabel, set: true }];

      describe('/label quick action', () => {
        describe('when the line starts with `/label`', () => {
          it('shows labels that are not currently selected', () => {
            const fullText = '/label ~';
            const selectionStart = 8;

            expect(labelsFilter({ collection, fullText, selectionStart })).toEqual([
              collection[0],
              collection[1],
            ]);
          });
        });

        describe('when the line does not start with `/label`', () => {
          it('shows all labels', () => {
            const fullText = '~';
            const selectionStart = 1;

            expect(labelsFilter({ collection, fullText, selectionStart })).toEqual(collection);
          });
        });
      });

      describe('/unlabel quick action', () => {
        describe('when the line starts with `/unlabel`', () => {
          it('shows labels that are currently selected', () => {
            const fullText = '/unlabel ~';
            const selectionStart = 10;

            expect(labelsFilter({ collection, fullText, selectionStart })).toEqual([collection[2]]);
          });
        });

        describe('when the line does not start with `/unlabel`', () => {
          it('shows all labels', () => {
            const fullText = '~';
            const selectionStart = 1;

            expect(labelsFilter({ collection, fullText, selectionStart })).toEqual(collection);
          });
        });
      });
    });
  });

  describe('members config', () => {
    const membersConfig = tributeConfig[GfmAutocompleteType.Members].config;
    const membersFilter = tributeConfig[GfmAutocompleteType.Members].filterValues;
    const userMember = {
      type: 'User',
      username: 'myusername',
      name: "My Name <script>alert('hi')</script>",
      avatar_url: '/uploads/-/system/user/avatar/123456/avatar.png',
      availability: null,
    };
    const groupMember = {
      type: 'Group',
      username: 'gitlab-com/support/1-1s',
      name: "GitLab.com / GitLab Support Team / 1-1s <script>alert('hi')</script>",
      avatar_url: null,
      count: 2,
      mentionsDisabled: null,
    };

    it('uses @ as the trigger', () => {
      expect(membersConfig.trigger).toBe('@');
    });

    it('inserts the username on autocomplete selection', () => {
      expect(membersConfig.fillAttr).toBe('username');
    });

    it('searches using both the name and username for a user', () => {
      expect(membersConfig.lookup(userMember)).toBe(`${userMember.name}${userMember.username}`);
    });

    it('searches using only its own name and not its ancestors for a group', () => {
      expect(membersConfig.lookup(groupMember)).toBe(last(groupMember.name.split(' / ')));
    });

    it('limits the items in the autocomplete menu to 10', () => {
      expect(membersConfig.menuItemLimit).toBe(10);
    });

    it('shows the avatar, name and username in the menu item for a user', () => {
      expect(membersConfig.menuItemTemplate({ original: userMember })).toMatchSnapshot();
    });

    it('shows an avatar character, name, parent name, and count in the menu item for a group', () => {
      expect(membersConfig.menuItemTemplate({ original: groupMember })).toMatchSnapshot();
    });

    describe('filter', () => {
      const assignees = [userMember.username];
      const collection = [userMember, groupMember];

      describe('/assign quick action', () => {
        describe('when the line starts with `/assign`', () => {
          it('shows members that are not currently selected', () => {
            const fullText = '/assign @';
            const selectionStart = 9;

            expect(membersFilter({ assignees, collection, fullText, selectionStart })).toEqual([
              collection[1],
            ]);
          });
        });

        describe('when the line does not start with `/assign`', () => {
          it('shows all labels', () => {
            const fullText = '@';
            const selectionStart = 1;

            expect(membersFilter({ assignees, collection, fullText, selectionStart })).toEqual(
              collection,
            );
          });
        });
      });

      describe('/unassign quick action', () => {
        describe('when the line starts with `/unassign`', () => {
          it('shows members that are currently selected', () => {
            const fullText = '/unassign @';
            const selectionStart = 11;

            expect(membersFilter({ assignees, collection, fullText, selectionStart })).toEqual([
              collection[0],
            ]);
          });
        });

        describe('when the line does not start with `/unassign`', () => {
          it('shows all members', () => {
            const fullText = '@';
            const selectionStart = 1;

            expect(membersFilter({ assignees, collection, fullText, selectionStart })).toEqual(
              collection,
            );
          });
        });
      });
    });
  });

  describe('merge requests config', () => {
    const mergeRequestsConfig = tributeConfig[GfmAutocompleteType.MergeRequests].config;
    const groupContextMergeRequest = {
      iid: 456789,
      reference: 'gitlab!456789',
      title: "Group context merge request title <script>alert('hi')</script>",
    };
    const projectContextMergeRequest = {
      id: null,
      iid: 123456,
      time_estimate: 0,
      title: "Project context merge request title <script>alert('hi')</script>",
    };

    it('uses ! as the trigger', () => {
      expect(mergeRequestsConfig.trigger).toBe('!');
    });

    it('searches using both the iid and title', () => {
      expect(mergeRequestsConfig.lookup(projectContextMergeRequest)).toBe(
        `${projectContextMergeRequest.iid}${projectContextMergeRequest.title}`,
      );
    });

    it('limits the number of rendered items to 100', () => {
      expect(mergeRequestsConfig.menuItemLimit).toBe(100);
    });

    it('shows the reference and title in the menu item within a group context', () => {
      expect(
        mergeRequestsConfig.menuItemTemplate({ original: groupContextMergeRequest }),
      ).toMatchSnapshot();
    });

    it('shows the iid and title in the menu item within a project context', () => {
      expect(
        mergeRequestsConfig.menuItemTemplate({ original: projectContextMergeRequest }),
      ).toMatchSnapshot();
    });

    it('inserts the reference on autocomplete selection within a group context', () => {
      expect(mergeRequestsConfig.selectTemplate({ original: groupContextMergeRequest })).toBe(
        groupContextMergeRequest.reference,
      );
    });

    it('inserts the iid on autocomplete selection within a project context', () => {
      expect(mergeRequestsConfig.selectTemplate({ original: projectContextMergeRequest })).toBe(
        `!${projectContextMergeRequest.iid}`,
      );
    });
  });

  describe('milestones config', () => {
    const milestonesConfig = tributeConfig[GfmAutocompleteType.Milestones].config;
    const milestone = {
      id: null,
      iid: 49,
      title: "13.2 <script>alert('hi')</script>",
    };

    it('uses % as the trigger', () => {
      expect(milestonesConfig.trigger).toBe('%');
    });

    it('searches using the title', () => {
      expect(milestonesConfig.lookup).toBe('title');
    });

    it('limits the number of rendered items to 100', () => {
      expect(milestonesConfig.menuItemLimit).toBe(100);
    });

    it('shows the title in the menu item', () => {
      expect(milestonesConfig.menuItemTemplate({ original: milestone })).toMatchSnapshot();
    });

    it('inserts the title on autocomplete selection', () => {
      expect(milestonesConfig.selectTemplate({ original: milestone })).toBe(
        `%"${escape(milestone.title)}"`,
      );
    });
  });

  describe('quick actions config', () => {
    const quickActionsConfig = tributeConfig[GfmAutocompleteType.QuickActions].config;
    const quickAction = {
      name: 'unlabel',
      aliases: ['remove_label'],
      description: 'Remove all or specific label(s)',
      warning: '',
      icon: '',
      params: ['~label1 ~"label 2"'],
    };

    it('uses / as the trigger', () => {
      expect(quickActionsConfig.trigger).toBe('/');
    });

    it('inserts the name on autocomplete selection', () => {
      expect(quickActionsConfig.fillAttr).toBe('name');
    });

    it('searches using both the name and aliases', () => {
      expect(quickActionsConfig.lookup(quickAction)).toBe(
        `${quickAction.name}${quickAction.aliases.join(', /')}`,
      );
    });

    it('limits the number of rendered items to 100', () => {
      expect(quickActionsConfig.menuItemLimit).toBe(100);
    });

    it('shows the name, aliases, params and description in the menu item', () => {
      expect(quickActionsConfig.menuItemTemplate({ original: quickAction })).toMatchSnapshot();
    });
  });

  describe('snippets config', () => {
    const snippetsConfig = tributeConfig[GfmAutocompleteType.Snippets].config;
    const snippet = {
      id: 123456,
      title: "Snippet title <script>alert('hi')</script>",
    };

    it('uses $ as the trigger', () => {
      expect(snippetsConfig.trigger).toBe('$');
    });

    it('inserts the id on autocomplete selection', () => {
      expect(snippetsConfig.fillAttr).toBe('id');
    });

    it('searches using both the id and title', () => {
      expect(snippetsConfig.lookup(snippet)).toBe(`${snippet.id}${snippet.title}`);
    });

    it('limits the number of rendered items to 100', () => {
      expect(snippetsConfig.menuItemLimit).toBe(100);
    });

    it('shows the id and title in the menu item', () => {
      expect(snippetsConfig.menuItemTemplate({ original: snippet })).toMatchSnapshot();
    });
  });
});
