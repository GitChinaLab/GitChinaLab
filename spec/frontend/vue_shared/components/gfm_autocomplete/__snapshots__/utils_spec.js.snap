// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gfm_autocomplete/utils emojis config shows the emoji name and icon in the menu item 1`] = `"raised_hands <gl-emoji data-name=\\"raised_hands\\"></gl-emoji>"`;

exports[`gfm_autocomplete/utils issues config shows the iid and title in the menu item within a project context 1`] = `"<small>123456</small> Project context issue title &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt;"`;

exports[`gfm_autocomplete/utils issues config shows the reference and title in the menu item within a group context 1`] = `"<small>gitlab#987654</small> Group context issue title &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt;"`;

exports[`gfm_autocomplete/utils labels config shows the title in the menu item 1`] = `
"
        <span class=\\"dropdown-label-box\\" style=\\"background: #123456;\\"></span>
        bug &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt;"
`;

exports[`gfm_autocomplete/utils members config shows an avatar character, name, parent name, and count in the menu item for a group 1`] = `
"
        <div class=\\"gl-display-flex gl-align-items-center\\">
          <div class=\\"gl-avatar gl-avatar-s32 gl-flex-shrink-0 gl-rounded-small
        gl-display-flex gl-align-items-center gl-justify-content-center\\" aria-hidden=\\"true\\">
            G</div>
          <div class=\\"gl-line-height-normal gl-ml-4\\">
            <div>1-1s &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt; (2)</div>
            <div class=\\"gl-text-gray-700\\">GitLab Support Team</div>
          </div>
          
        </div>
      "
`;

exports[`gfm_autocomplete/utils members config shows the avatar, name and username in the menu item for a user 1`] = `
"
        <div class=\\"gl-display-flex gl-align-items-center\\">
          <img class=\\"gl-avatar gl-avatar-s32 gl-flex-shrink-0 gl-avatar-circle\\" src=\\"/uploads/-/system/user/avatar/123456/avatar.png\\" alt=\\"\\" />
          <div class=\\"gl-line-height-normal gl-ml-4\\">
            <div>My Name &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt;</div>
            <div class=\\"gl-text-gray-700\\">@myusername</div>
          </div>
          
        </div>
      "
`;

exports[`gfm_autocomplete/utils merge requests config shows the iid and title in the menu item within a project context 1`] = `"<small>123456</small> Project context merge request title &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt;"`;

exports[`gfm_autocomplete/utils merge requests config shows the reference and title in the menu item within a group context 1`] = `"<small>gitlab!456789</small> Group context merge request title &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt;"`;

exports[`gfm_autocomplete/utils milestones config shows the title in the menu item 1`] = `"13.2 &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt;"`;

exports[`gfm_autocomplete/utils quick actions config shows the name, aliases, params and description in the menu item 1`] = `
"<div>/unlabel <small>(or /remove_label)</small> <small>~label1 ~\\"label 2\\"</small></div>
          <div><small><em>Remove all or specific label(s)</em></small></div>"
`;

exports[`gfm_autocomplete/utils snippets config shows the id and title in the menu item 1`] = `"<small>123456</small> Snippet title &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt;"`;
